### PR TITLE
Make using cache for last_transition optional

### PR DESCRIPTION
**Pull request by petehamilton**
@greysteil What do you think? Need to add specs for other adapters but it's a first cut.

### CODE REVIEW

File: lib/statesman/adapters/memory.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def create(from, to, metadata = {})
         transition
       end
 
-      def last
+      def last(use_cache: true)
```
Comment (petehamilton) [inline lib/statesman/adapters/memory.rb]: Not hugely keen on this. Be nice if it wasn't an unused arg...

Comment (greysteil) [inline lib/statesman/adapters/memory.rb]: I think that's fine, but I'm not sure about `use_cache`. Would rather have busting the cache as the explicit action, so something like `force_reload` (although that's not perfect).

File: spec/statesman/adapters/active_record_spec.rb [hunk 2]
```diff
@@ -207,12 +207,32 @@
...
+            to receive(:my_active_record_model_transitions).never
+          expect(adapter.last.to_state).to eq("y")
+        end
+
+        context "when explitly not using the cache" do
+          it "still returns the cached value" do
+            expect(adapter.last(use_cache: false).to_state).to eq("z")
+          end
+        end
+      end
```
Comment (greysteil) [inline spec/statesman/adapters/active_record_spec.rb]: Can we add a shared adapter spec, too, so that all adapters have the same method signature?

**Comment by greysteil** [comment]
:+1: after naming.

File: lib/statesman/adapters/memory.rb (hunk 1 above)
Comment (petehamilton) [inline lib/statesman/adapters/memory.rb]: I agree - making 'bust' be the affirmative makes more sense to me.

**Pull request merged**